lms/djangoapps/api_manager/sessions/views.py

```diff
@@ -120,6 +121,10 @@ def post(self, request):
                     response_data['uri'] = '{}/{}'.format(base_uri, new_session.session_key)
                     response_status = status.HTTP_201_CREATED
 
+                    # generate a CSRF tokens for any web clients that may need to
+                    # call into the LMS via Ajax (for example Notifications)
+                    response_data['csrftoken'] = RequestContext(request, {}).get('csrf_token')
```

Comment:
@mattdrayer FYI, I had to generate a csrftoken on login because I found cases where - in McKA - this cookie was not getting set until the end-user happened to view an xBlock, which - in turn - sets the cookie. Now, I'm returning a csrftoken to Apros which is setting the cookie for the end-user.